qa/agent_g4_row.py

```
@@ -0,0 +1,72 @@
+                   f"route={args.route_complete} pass1={args.pass1} pass2={args.pass2}."),
+    }
+    for key in ("legibility_median", "actor_luminance_floor"):
+        if getattr(args, key) is not None:
+            row[key] = getattr(args, key)
+    if args.frames_per_room:
+        try:
+            row["frames_per_room"] = json.loads(args.frames_per_room)
+        except json.JSONDecodeError as exc:
+            raise ValueError("--frames-per-room must be a JSON object") from exc
+        if not isinstance(row["frames_per_room"], dict):
+            raise ValueError("--frames-per-room must be a JSON object")
```

Ruff (0.16.3)

[warning] Avoid specifying long messages outside the exception class

(TRY003)